lib/ansible/modules/cloud/amazon/ec2_group.py

```diff
@@ -448,179 +445,252 @@ def main():
             if curGroup.name == name and (vpc_id is None or curGroup.vpc_id == vpc_id):
                 group = curGroup
 
-    # Ensure requested group is absent
-    if state == 'absent':
-        if group:
-            # found a match, delete it
-            try:
-                if not module.check_mode:
-                    group.delete()
-            except BotoServerError as e:
-                module.fail_json(msg="Unable to delete security group '%s' - %s" % (group, e.message), exception=traceback.format_exc())
-            else:
-                group = None
-                changed = True
-        else:
-            # no match found, no changes required
-            pass
+    return group, groups
 
-    # Ensure requested group is present
-    elif state == 'present':
-        if group:
-            # existing group
-            if group.description != description:
-                module.fail_json(msg="Group description does not match existing group. ec2_group does not support this case.")
 
-        # if the group doesn't exist, create it now
-        else:
-            # no match found, create it
+def remove_group(module, ec2):
+    (group, groups) = lookup_groups(module, ec2)
+
+    if group:
+        try:
             if not module.check_mode:
-                group = ec2.create_security_group(name, description, vpc_id=vpc_id)
+                group.delete()
+        except BotoServerError as e:
+            module.fail_json(msg="Unable to delete security group '%s' - %s" % (group, e.message), exception=traceback.format_exc())
+        return True, None
+    else:
+        return False, None
+
+
+def create_group(module, ec2):
+    name = module.params['name']
+    description = module.params['description']
+    vpc_id = module.params['vpc_id']
+
+    if not module.check_mode:
+        group = ec2.create_security_group(name, description, vpc_id=vpc_id)
+
+        # When a group is created, an egress_rule ALLOW ALL
+        # to 0.0.0.0/0 is added automatically but it's not
+        # reflected in the object returned by the AWS API
+        # call. We re-read the group for getting an updated object
+        # amazon sometimes takes a couple seconds to update the security group so wait till it exists
+        while len(ec2.get_all_security_groups(filters={'group_id': group.id})) == 0:
+            time.sleep(0.1)
```

Comment:
Hi @ryansb, thanks for the review. So because of the AWSRetry, I guess I will close the PR and wait tfor the #25340 to be merged. Is there something I can do to make it sooner rather then later?